CMakeLists.txt

```diff
@@ -161,6 +161,9 @@ endif ()
 if (FMT_PEDANTIC)
   target_compile_options(fmt PRIVATE ${PEDANTIC_COMPILE_FLAGS})
 endif ()
+if (MSVC)
+  target_compile_options(fmt PRIVATE /EHsc)
```

Comment:
Exceptions are enabled by default on MSVC. Why is this needed?

Reply:
This would also force the library into a specific exception handling model, which would be undesirable.

Reply:
@vitaut : My clang-target-msvc isn't defaulting /EHsc. Then I guess this should just trying to mirror /EHsc default in MSVC cl.exe. Perhaps...

if (MSVC)
should be changed into
if (MSVC AND CMAKE_CXX_COMPILER_ID MATCHES "Clang" )

@eliaskosunen Would you be okay with narrowing to clang-target-msvc for mirroring the default /EHsc above?

Should there be FMT_MSVC_EH option that explicitly add custom /EHxx?

Reply:
I think, perhaps we can have FMT_MSVC_EH like this :

```
if (MSVC)
  if ( (NOT FMT_MSVC_EH) AND (CMAKE_CXX_COMPILER_ID MATCHES "Clang") )
    set(FMT_CUSTOM_MSVC_EH /EHsc)
  else ()
    set(FMT_CUSTOM_MSVC_EH ${FMT_MSVC_EH})
  endif ()
  target_compile_options(fmt PRIVATE ${FMT_CUSTOM_MSVC_EH})
endif ()
```

What do you guys think?

Reply:
I don't know why TravisCI evaluates MSVC true in apple platform.
So I try to workaround with

```
if (MSVC)
  if (FMT_MSVC_EH)
    set(FMT_CUSTOM_MSVC_EH ${FMT_MSVC_EH})
  else ()
    if ( (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
	        AND NOT (CMAKE_CXX_COMPILER_ID MATCHES "AppleClang") )
      set(FMT_CUSTOM_MSVC_EH /EHsc)
    else ()
      set(FMT_CUSTOM_MSVC_EH)
    endif ()
  endif ()
  target_compile_options(fmt PRIVATE ${FMT_CUSTOM_MSVC_EH})
endif ()
```

Reply:
It seems that it's because my excessing endif, not MSVC weirdness. Fixed it.

Reply:
> My clang-target-msvc isn't defaulting /EHsc.

So what is the default in clang-target-msvc? Exceptions disabled or enabled with a different setting?

Reply:
It seems exceptions are disabled by default. Well, you are right, after take a look at => [user manual](https://clang.llvm.org/docs/UsersManual.html#configuration-files), I found that users can explicitly state the configuations through very generous means. However, `clang-cl.exe`'s alone basically not enable exceptions by default.

If you think this compiler case wouldn't be worth for the added complexity, I will be fine with dropping these CMAKE changes for /EHsc.

Reply:
I've removed conditional adding /EHsc from CMAKELists.txt in both root and test.